EPI/parse-nested-GrFN.py

```diff
@@ -0,0 +1,124 @@
+import json
+from argparse import ArgumentParser
+
+
+def formatGraph(data): 
+    modelMetadata = {}
+    nodesDict = {}
+    nodes = []
+    edges = []
+
+    for key,value in data.items():
+        if key == 'variables':
+            for item in value: 
+                if 'metadata' in item:
+                    metadata = item['metadata'][0]
+                else:
+                    metadata = {}
+                item['nodeType'] = 'variable'
+                item['metadata'] = metadata
+                nodesDict[item['uid']] = item
+        if key == 'functions':
+            for item in value: 
+                if 'metadata' in item:
+                    metadata = item['metadata'][0]
+                else:
+                    metadata = {} 
+                item['nodeType'] = 'function'
+                item['metadata'] = metadata
+                nodesDict[item['uid']] = item
+        if key == 'hyper_edges':
+            for item in value: 
+                if len(item['inputs']) > 0:
+                    for i in item['inputs']: # Inputs array is sometimes empty
+                        first_edge = { 'source': i, 'target': item['function']}
+                        edges.append(first_edge)
+                
+                for o in item['outputs']:
+                    second_edge = { 'source': item['function'], 'target': o }
+                    edges.append(second_edge)
+        if key == 'subgraphs':
+            for item in value: 
+                # Get parent name
+                parent_name = item['scope']
+                if (parent_name == '@global'):
+                    parent_name = 'root'
+
+                # Get container name
+                splitted_id = item['basename'].split('.')
+
+                #Get container metadata
+                if 'metadata' in item:
+                    metadata = item['metadata'][0]
+
+                node = { 'id': item['basename'], 'concept': splitted_id[(len(splitted_id) - 1)], 'nodeType': item['type'], 'label': splitted_id[(len(splitted_id) - 1)], 'parent': parent_name, 'metadata': metadata }
+                nodes.append(node)
+                for n in item['nodes']:
+                    found = nodesDict[n]
```

Comment:
should we check if `n` is in `nodesDict` first, and raise a meaningful exception if it isn't? I assume this would be a bug in the GrFN if the node was missing.

Reply:
Done.